### ISSUE TITLE
Add Test Module Command unavailable in unrecognized hosts - HostSupportsUnitTests

### ISSUE THREAD

**Issue by ThunderFrame** [rubberduck-vba/Rubberduck#3854]
Now that Unit Tests are supported in all VBA hosts, it seems we forgot to allow adding Test Modules in all VBA hosts. It is possible to add the annotations and stubs yourself (after which (and a reparse) you can add test methods), but it obviously makes sense to allow adding a Test Module, for any VBA host

The `HostSupportsUnitTests` condition needs to be dropped, and as the condition is only used in a single location, the method can probably be removed.

I'm unsure why `HostSupportsUnitTests` is defined as an extension method in `Rubberduck.VBEEditor/Extensions/IDEExtensions.cs`, but also implemented in `Rubberduck.VBEEditor/SafeComWrappers/VBA/VBE.cs`

**Comment by bclothier** [comment]
I suspect it was moved to the extensions so that it can support both VBA and VB6 but the original code in the `VBA.VBE` wasn't removed. The `VB6.VBE` apparently never had it.

That said, I agree that this should be removed but because we also rely on a different method, it makes sense to test that new method. 

This is my tentative solution which I will introduce in a separate PR once the other PR for installer is merged. Basically on the first use, we check that we can create a component and then run code. Those are 2 prerequisites for us to be able to create unit tests; if we can't create modules, then we can't create test module. Likewise if we can't run the code, then we can't use unit tests:

```

        protected override bool EvaluateCanExecute(object parameter)
        {
            var project = GetProject();
            return !project.IsWrappingNullReference && CanExecuteCode(project);
        }

        private Dictionary<string, bool> _canProjectExecuteCode = new Dictionary<string, bool>();
        private bool CanExecuteCode(IVBProject project)
        {
            if (!_canProjectExecuteCode.ContainsKey(project.Name))
            {
                _uiDispatcher.Invoke(() =>
                {
                    IVBComponent temp = null;
                    try
                    {

                        var tempCode = $@"
Public Function TestExecuteCode As Boolean
  TestExecuteCode = True
End Function
";
                        var typeLib = new VBETypeLibsAPI();
                        temp = project.VBComponents.Add(ComponentType.StandardModule);
                        temp.CodeModule.AddFromString(tempCode);
                        if (typeLib.CompileComponent(_vbe, project.Name, temp.Name))
                        {
                            var result = typeLib.ExecuteCode(temp, "TestExecuteCode");
                            _canProjectExecuteCode.Add(project.Name, Convert.ToBoolean(result));
                        }
                    }
                    catch (Exception ex)
                    {
                        Logger.Warn("Unable to execute code with the current host; assuming unit tests not supported.");
                        Logger.Warn(ex);
                        _canProjectExecuteCode.Add(project.Name, false);
                    }
                    finally
                    {
                        if (temp != null && !temp.IsWrappingNullReference)
                        {
                            project.VBComponents.Remove(temp);
                            temp.Dispose();
                        }
                    }
                });
            }

            return _canProjectExecuteCode[project.Name];
        }
```
